pkg/spec3/fuzz.go

```diff
@@ -0,0 +1,250 @@
+			c.Fuzz(&val)
+			v.Paths["/"+c.RandString()] = &val
+		}
+	},
+	func(v *SecurityScheme, c fuzz.Continue) {
+		if c.Intn(refChance) == 0 {
+			c.Fuzz(&v.Refable)
+			return
+		}
+		switch c.Intn(4) {
+		case 0:
+			v.Type = "apiKey"
+			v.Name = c.RandString() + "x"
+			switch c.Intn(3) {
+			case 0:
+				v.In = "query"
+			case 1:
+				v.In = "header"
+			case 2:
+				v.In = "cookie"
+			}
+		case 1:
+			v.Type = "http"
+		case 2:
+			v.Type = "oauth2"
+			v.Flows = make(map[string]*OAuthFlow)
+			flow := OAuthFlow{}
+			flow.AuthorizationUrl = c.RandString() + "x"
+			v.Flows["implicit"] = &flow
+			flow.Scopes = make(map[string]string)
+			flow.Scopes["foo"] = "bar"
+		case 3:
+			v.Type = "openIdConnect"
+			v.OpenIdConnectUrl = "https://" + c.RandString()
+		}
+		v.Scheme = "basic"
+	},
+	func(v *spec.Ref, c fuzz.Continue) {
+		switch c.Intn(7) {
+		case 0:
+			*v = spec.MustCreateRef("#/components/schemas/" + randAlphanumString())
+		case 1:
+			*v = spec.MustCreateRef("#/components/responses/" + randAlphanumString())
+		case 2:
+			*v = spec.MustCreateRef("#/components/headers/" + randAlphanumString())
+		case 3:
+			*v = spec.MustCreateRef("#/components/securitySchemes/" + randAlphanumString())
+		case 5:
+			*v = spec.MustCreateRef("#/components/parameters/" + randAlphanumString())
+		case 6:
+			*v = spec.MustCreateRef("#/components/requestBodies/" + randAlphanumString())
+		}
+	},
+	func(v *Parameter, c fuzz.Continue) {
+		if c.Intn(refChance) == 0 {
+			c.Fuzz(&v.Refable)
+			return
+		}
+		c.Fuzz(&v.ParameterProps)
+		c.Fuzz(&v.VendorExtensible)
+
+		switch c.Intn(3) {
+		case 0:
+			// Header param
+			v.In = "query"
+		case 1:
+			v.In = "header"
+		case 2:
+			v.In = "cookie"
+		}
+	},
+	func(v *RequestBody, c fuzz.Continue) {
+		if c.Intn(refChance) == 0 {
+			c.Fuzz(&v.Refable)
+			return
+		}
+		c.Fuzz(&v.RequestBodyProps)
+		c.Fuzz(&v.VendorExtensible)
+	},
+	func(v *Header, c fuzz.Continue) {
+		if c.Intn(refChance) == 0 {
+			c.Fuzz(&v.Refable)
+			return
+		}
+		c.Fuzz(&v.HeaderProps)
+		c.Fuzz(&v.VendorExtensible)
+	},
+	func(v *ResponsesProps, c fuzz.Continue) {
+		c.Fuzz(&v.Default)
+		n := c.Intn(5)
+		for i := 0; i < n; i++ {
+			r2 := Response{}
+			c.Fuzz(&r2)
+			// HTTP Status code in 100-599 Range
+			code := c.Intn(500) + 100
+			v.StatusCodeResponses = make(map[int]*Response)
+			v.StatusCodeResponses[code] = &r2
+		}
+	},
+	func(v *Response, c fuzz.Continue) {
+		if c.Intn(refChance) == 0 {
+			c.Fuzz(&v.Refable)
+			return
+		}
+		c.Fuzz(&v.ResponseProps)
+		c.Fuzz(&v.VendorExtensible)
+	},
+	func(v *spec.Extensions, c fuzz.Continue) {
+		*v = spec.Extensions{}
+		numChildren := c.Intn(5)
+		if numChildren == 0 {
+			*v = nil
+		}
+		for i := 0; i < numChildren; i++ {
+			v.Add("x-"+randAlphanumString(), c.RandString()+"x")
+		}
+	},
+	func(v *spec.ExternalDocumentation, c fuzz.Continue) {
+		c.Fuzz(&v.Description)
+		v.URL = "https://" + randAlphanumString()
+	},
+	func(v *spec.Schema, c fuzz.Continue) {
```

Comment:
I have only checked quickly but this is missing a lot of fields, which means that if these fields are not round-tripped properly, we're very likely to miss that in the tests.

Reply:
Let me see if we can reuse the fuzzer that Alex has already wrote for this.